addons/rating/tests/test_controller.py

```diff
@@ -34,11 +34,13 @@ def test_controller_rating(self):
         req = self.url_open(url)
         self.assertEqual(req.status_code, 200, "Response should = OK")
 
+        # changed behavior in Odoo 16+: the GET request to /rate/{access_token}/int
+        # will not trigger a consume of the rating. User needs to submit the Form
         details = [
             (self.user_demo.login, rating_test_1, rating_test_1.access_token, False),
-            (self.user.login, rating_test_1, rating_test_1.access_token, True),
-            (None, rating_test_2, rating_test_2.access_token, True),
-            (self.user_portal.login, rating_test_3, rating_test_3.access_token, True)
+            (self.user.login, rating_test_1, rating_test_1.access_token, False),
+            (None, rating_test_2, rating_test_2.access_token, False),
+            (self.user_portal.login, rating_test_3, rating_test_3.access_token, False)
```

Comment:
@tde-banana-odoo 
New FW, new broken test.

Seems like this test mostly check if a logged user is not supposed to be able to rate a record, even with access token.
It assumed that the rating would be consumed on the initial GET. Adapted it to now expect consumed == False. No it will mostly test if the response.content contains a given string in a given scenario.

So unless you don't agree with this, we should be green for a r+

Reply:
Indeed, the "right" test is below (checking that 'You cannot rate this'). Could be improved to make test easier to read and understand but hey, out of scope for this PR :) .